From e549f55bb90c2da0e2a4d7e9ae82151548a7d941 Mon Sep 17 00:00:00 2001
From: Rakes Pani <rakes.pani@lge.com>
Date: Fri, 22 Jan 2021 09:50:37 +0530
Subject: [PATCH] Add CIDR prefix length as ipv4 property

:Release Notes:
Add CIDR prefix length as ipv4 property

:Detailed Notes:
Add CIDR prefix length as ipv4 property

:Testing Performed:
Built and Tested

:QA Notes:
NA

:Issues Addressed:
[PLAT-123931] Implementation for addition of ip rules to
              maintain multiple Wired interfaces list in connman

Upstream-Status: Inappropriate[webos specific]
---
 src/ipconfig.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/ipconfig.c b/src/ipconfig.c
index 08576820..fa9def70 100644
--- a/src/ipconfig.c
+++ b/src/ipconfig.c
@@ -1919,6 +1919,10 @@ void __connman_ipconfig_append_ipv4(struct connman_ipconfig *ipconfig,
 		connman_dbus_dict_append_basic(iter, "Address",
 				DBUS_TYPE_STRING, &append_addr->local);
 
+		connman_dbus_dict_append_basic(iter, "PrefixLength",
+						DBUS_TYPE_BYTE,
+						&append_addr->prefixlen);
+
 		addr = 0xffffffff << (32 - append_addr->prefixlen);
 		netmask.s_addr = htonl(addr);
 		mask = inet_ntoa(netmask);
